src/components/filter/filterComposition.tsx

```diff
@@ -0,0 +1,128 @@
+import {
+  FormSelect,
+  FormSelectOption,
+  TextInput,
+  ToolbarGroup,
+  ToolbarItem,
+} from '@patternfly/react-core';
+import React from 'react';
+import { InjectedTranslateProps, translate } from 'react-i18next';
+import SelectFilter from './selectFilter';
+
+type FilterKind = 'type' | 'name';
+type BufferUpdater = (current: { name: string; value: string }) => void;
+
+interface Props extends InjectedTranslateProps {
+  query: Query;
+  filters: FilterKind[];
+  name: string;
+  value: string;
+  updateFilter: BufferUpdater;
+  switchType: (current: { name: string; value: string }) => void;
+  onSearch: (newQuery: { [k: string]: string }) => void;
+}
+
+interface Query {
+  [k: string]: string;
+}
+interface Filter {
+  name: string;
+  value: string;
+}
+type Mutate = (query: Query, filter: Filter) => Query;
+
+const addMultiValue = (query: Query, buffer: Filter) => {
+  let newValue = buffer.value;
+  if (buffer.name === 'Name') {
+    newValue = query[buffer.name]
+      ? [query[buffer.name], buffer.value].join(',')
+      : buffer.value;
+  }
+  return {
+    ...query,
+    [buffer.name]: newValue,
+  };
+};
+
+const searchOnEnter = (
+  query: Query,
+  filter: Filter,
+  mutate: Mutate,
+  onSearch: (newQuery: Query) => void
+) => (event: React.KeyboardEvent) => {
+  if (event.key === 'Enter' && filter.value) {
+    onSearch(mutate(query, filter));
+  }
+};
+const FilterCompositionBase: React.SFC<Props> = ({
+  query,
+  filters,
+  name,
+  value,
+  updateFilter,
+  switchType,
+  onSearch,
+  t,
+}) => {
+  const filterController =
+    name === 'Type' ? (
+      <FormSelect
+        aria-label={t('filter.type_aria_label')}
+        value={name}
+        onChange={newValue => onSearch({ name, value: newValue })}
+      >
+        <FormSelectOption
+          key={`type-option-empty`}
+          value={''}
+          label={t('filter.type_empty')}
+        />
+        <FormSelectOption
+          key={`type-option-openshift`}
+          value={'OCP'}
+          label={t('filter.type_ocp')}
+        />
+        <FormSelectOption
+          key={`type-option-aws`}
+          value={'AWS'}
+          label={t('filter.type_aws')}
+        />
+      </FormSelect>
+    ) : (
+      <TextInput
+        value={value}
+        placeholder={t('source_details.filter.placeholder', {
+          value: name,
+        })}
+        id="sources filter value"
```

Comment:
I'm thinking this should be located under the cost models directory. I don't see this as a reusable component, unless you can make it more generic? For example, the ID is specific to sources. And the only options are 'OCP' and 'AWS'.

Reply:
Yes, I should move this under costModelsDetails and later make it more generic.

I will open a ticket for refactoring it and use it in all the tables we have.